# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: netwerk/base/nsSocketTransportService2.cpp
# Commit: a0e755ca9005
# Full Hash: a0e755ca9005618be5c4d9d667d2c316a4032d30
# Author: Randell Jesup <rjesup@mozilla.com>
# Date: 2023-03-27 21:11:43
# Description:
#   Bug 1824152: Return success if SocketThreadService was already shutting down r=necko-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D173634
# ==============================================================================

diff -r acea5a392b06 -r a0e755ca9005 netwerk/base/nsSocketTransportService2.cpp
--- a/netwerk/base/nsSocketTransportService2.cpp	Mon Mar 27 14:54:30 2023 +0000
+++ b/netwerk/base/nsSocketTransportService2.cpp	Mon Mar 27 14:54:56 2023 +0000
@@ -809,14 +809,11 @@
 
   NS_ENSURE_STATE(NS_IsMainThread());
 
-  if (!mInitialized) {
+  if (!mInitialized || mShuttingDown) {
+    // We never inited, or shutdown has already started
     return NS_OK;
   }
 
-  if (mShuttingDown) {
-    return NS_ERROR_UNEXPECTED;
-  }
-
   {
     auto observersCopy = mShutdownObservers;
     for (auto& observer : observersCopy) {
